From c84ff5f29ce36e1b5f253ab08cdb8e8952fa3f26 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mos=C3=A8=20Giordano?= <mose@gnu.org>
Date: Tue, 11 Nov 2025 17:20:16 -0800
Subject: [PATCH] GCC 13: Increase fallback min macOS version for libgcc to
 10.8

`-mmacosx-version-min=10.5` seems to cause problems with ld64 from cctools
(https://github.com/JuliaPackaging/Yggdrasil/pull/10132#issuecomment-2569925646),
and in any case in Julia/BinaryBuilder we support macOS 10.12 at minimum.

Backport of GCC 14 patch to 13
---
 libgcc/config.host | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/libgcc/config.host b/libgcc/config.host
index 9aa36bf2210..82ef101f890 100644
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -239,10 +239,10 @@ case ${host} in
       tmake_file="t-darwin-min-8 $tmake_file"
       ;;
     *-*-darwin1[67]]*)
-      tmake_file="t-darwin-min-5 $tmake_file"
+      tmake_file="t-darwin-min-8 $tmake_file"
       ;;
     *-*-darwin9* | *-*-darwin1[0-5]*)
-      tmake_file="t-darwin-min-5 t-darwin-libgccs1 $tmake_file"
+      tmake_file="t-darwin-min-8 t-darwin-libgccs1 $tmake_file"
       ;;
     *-*-darwin[4-8]*)
       tmake_file="t-darwin-min-1 t-darwin-libgccs1 $tmake_file"
@@ -250,8 +250,8 @@ case ${host} in
     *)
       # Fall back to configuring for the oldest system known to work with
       # all archs and the current sources.
-      tmake_file="t-darwin-min-5 t-darwin-libgccs1 $tmake_file"
-      echo "Warning: libgcc configured to support macOS 10.5" 1>&2
+      tmake_file="t-darwin-min-8 t-darwin-libgccs1 $tmake_file"
+      echo "Warning: libgcc configured to support macOS 10.8" 1>&2
       ;;
   esac
   extra_parts="crt3.o libd10-uwfef.a crttms.o crttme.o libemutls_w.a"
-- 
2.45.2

